core/src/main/java/com/amplitude/core/Amplitude.kt

```diff
@@ -183,6 +184,18 @@ open class Amplitude internal constructor(
         return this
     }
 
+    /**
```

Comment:
Do we need `reset` in core?

Reply:
I think this is an open question. @qingzhuozhen feel free to suggest more. My understanding is `core` provides a basic implementation for all functions without the environment specific context. Let me know if this is wrong or you have any concerns!

Reply:
That is a good question. I guess we could keep this in core if we have the user id and device id method. @kevinpagtakhan @justin-fiedler any thoughts?

If we have the method in core, do we still need the android to have a different version?

Reply:
Feels in the core we cannot get the android context? 
initializeDeviceId method is only in androdiContextPlugin now. there is no such method in setup in the ContextPlugin

Reply:
Yes, core doesn't have the Android context. I don't want to introduce the Android dependency for core, imaging the code is running inside a Java server. Two ways list below and I pick up the min change way (second):
- Introduce a new plugin (e.g., ContextPlugin) with reset method, then core can use timeline to call all reset method of these plugins
- Core has a basic implementation, different context might override depending on runtime env